src/server/protocol.ts

```diff
@@ -2675,13 +2675,18 @@ export interface InlayHintsRequest extends Request {
 }
 
 export interface InlayHintItem {
-    text: string;
+    text: string | InlayHintItemDisplayPart[];

```

Comment:
We usually try to preserve backward-compatibility with older clients by making this functionality opt-in. So typically this would be done when through a `configure` request, or a specific property on the inlayHint request.

Reply:
I've added an user preference to enable the feature.